server/vault.api.js

```diff
@@ -0,0 +1,128 @@
+const express = require('express');
+const asyncHandler = require('express-async-handler');
+const HomebrewModel = require('./homebrew.model.js').model;
+
+const router = express.Router();
+
+const buildTitleConditions = (title) => {
+    if (!title) return {};
+    return {
+        $text: {
+            $search: title,
+            $caseSensitive: false,
+        },
+    };
+};
+
+const buildAuthorConditions = (author) => {
+    if (!author) return {};
+    return { authors: author };
+};
+
+//"$and": [ {"published": true}, {"$text": { "$search": "titleString", "$caseSensitive": false } }, { "authors" : "authorString"}]
+//is a good example of a query constructed with this function
+
+const handleErrorResponse = (res, error, functionName) => {
+    const status = error.response?.status || 500;
+    const message =
+        status === 503 ? 'Service Unavailable' : 'Internal Server Error';
+
+    console.error(`Error in ${functionName}:`, error);
+
+    return res.status(status).json({
+        errorCode: status.toString(),
+        message: `Error in function ${functionName}: ${message}`,
+    });
+};
+
+const buildBrewsQuery = (legacy, v3) => {
```

Comment:
Would it make more sense to rename this "buildRendererConditions" and just put the `published : true` down in the `combinedQuery` since it's always there by default? Let this function handle just one field?

Reply:
It would make sense too, yeah